lib/puppet/type/file/content.rb

```diff
@@ -217,6 +217,22 @@ def get_from_source(source_or_content, &block)
       end
     end
 
+    def get_from_http_source(source, &block)
+      uri = URI(source)
+      use_ssl = uri.scheme == 'https'
+      connection = Puppet::Network::HttpPool.http_instance(uri.host, uri.port, use_ssl)
+      connection.request_get(uri.path, add_accept_encoding({"Accept" => "raw"}), &block)
```

Comment:
What conditions cause us to come through this code path (which doesn't do an indirector `find`), vs. the code path in `generic_http` where we are doing an HTTP request as part of the `find`?